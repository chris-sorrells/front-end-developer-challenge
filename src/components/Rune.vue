<template>
  <a
    class="rune"
    :class="[{ active }, { toggleable }, `is-${type}`]"
    @contextmenu.prevent
    @click.left="$emit('activate')"
    @click.right="$emit('deactivate')"
  />
</template>

<script>
export default {
  props: {
    active: {
      type: Boolean,
      default: false
    },

    toggleable: {
      type: Boolean,
      default: false
    },

    type: {
      type: String,
      required: true,
      validator(value) {
        return [
          "box",
          "silverware",
          "cake",
          "crown",
          "boat",
          "goggles",
          "lightning",
          "skull"
        ].includes(value);
      }
    }
  }
};
</script>

<style lang="scss">
.rune {
  display: block;
  background: url("../assets/talent-icons-sprite.png") no-repeat;
  width: 50px;
  height: 50px;
  cursor: pointer;

  // &.toggleable {
  //   cursor: default;
  // }

  &.is-box {
    background-position: 0 -50px;
    &.active {
      background-position: 0 0;
    }
  }
  &.is-silverware {
    background-position: -50px -50px;
    &.active {
      background-position: -50px 0;
    }
  }
  &.is-cake {
    background-position: -100px -50px;
    &.active {
      background-position: -100px 0;
    }
  }
  &.is-crown {
    background-position: -150px -50px;
    &.active {
      background-position: -150px 0;
    }
  }
  &.is-boat {
    background-position: -200px -50px;
    &.active {
      background-position: -200px 0;
    }
  }
  &.is-goggles {
    background-position: -250px -50px;
    &.active {
      background-position: -250px 0;
    }
  }
  &.is-lightning {
    background-position: -300px -50px;
    &.active {
      background-position: -300px 0;
    }
  }
  &.is-skull {
    background-position: -350px -50px;
    &.active {
      background-position: -350px 0;
    }
  }
}
</style>
