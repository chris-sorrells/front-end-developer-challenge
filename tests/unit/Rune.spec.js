import { shallowMount } from "@vue/test-utils";
import Rune from "@/components/Rune.vue";

describe("Rune.vue", () => {
  // TODO
  it("renders", () => {
    const wrapper = shallowMount(Rune);
    expect(wrapper.element).toMatchSnapshot();
  });
});
