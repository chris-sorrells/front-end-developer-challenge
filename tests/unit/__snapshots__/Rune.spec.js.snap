// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue renders 1`] = `
<a>
  todo rune
</a>
`;
